src/test/scala/com/cognite/sdk/scala/common/OAuth2ClientCredentialsTest.scala

```diff
@@ -113,14 +114,14 @@ class OAuth2ClientCredentialsTest extends AnyFlatSpec with Matchers with OptionV
   it should "refresh tokens when they expire" in {
     import sttp.client3.impl.cats.implicits._
 
-    var numTokenRequests = 0
+    val numTokenRequests = new AtomicLong(0L)
```

Comment:
Using `Ref` would be preferable, https://typelevel.org/cats-effect/docs/std/ref#concurrent-counter
Then you won't need the manual `IO` wrappings.

Reply:
Thanks, I update using Ref 😁